app/controllers/api/samples_controller.rb

```diff
@@ -2,35 +2,35 @@ class Api::SamplesController < ApplicationController
   before_action :set_user, only: [:show, :update]
   
   def index
-    @users = User.all
-    render json: @users
+    users = User.all
+    render json: users
   end
 
   def show
-    render json: @user
+    render json: user
   end
 
   def create
-    @user = User.new(user_params)
-    if @user.save
-      render json: @user, status: :created
+    user = User.new(user_params)
+    if user.save
+      render json: user, status: :created
     else
-      render json: @user.errors, status: :unprocessable_entity
+      render json: user.errors, status: :unprocessable_entity
     end
   end
 
   def update
-    if @user.update(user_params)
-      render json: @user
+    if user.update(user_params)
+      render json: user
     else
-      render json: @user.errors, status: :unprocessable_entity
+      render json: user.errors, status: :unprocessable_entity
     end
   end
   
   private
 
   def set_user
-    @user = User.find(params[:id])
+    user = User.find(params[:id])
```

Comment:
```suggestion
    @user = User.find_by(id: params[:id])
```

This is a combination of two suggestions. First, we do want to keep the instance variable `@user`, otherwise it will only be set within `set_user`, and won't be available outside of it. `@users` was being created as an instance variable but used only in the `#index` action, so should be kept a regular variable, to avoid possible interactions. Please change all uses of `user` accordingly.

Secondly, and this is something you may not have learned, if you do a `find` for an object that doesn't exist, it returns an `ActiveRecord::RecordNotFound` error. This is sometimes ok, like if the id comes from Rails, but in this case one way we can get around that is to use `find_by` instead, which returns `nil` if `find` returns nothing.